hera_cal/frf.py

```diff
@@ -16,11 +16,85 @@
+SPEED_OF_LIGHT = const.c.si.value
+SDAY_KSEC = 86163.93 / 1000.
+
+
+def sky_frates(uvd, keys=None, frate_standoff=0.0, frate_width_multiplier=1.0, min_frate_half_width=0.025):
+    """Automatically compute sky fringe-rate ranges based on baselines and telescope location.
+
+    Parameters
+    ----------
+    uvd: UVData object
+        uvdata object of data to compute sky-frate limits for.
+    keys: list of antpairpol tuples, optional
+        list of antpairpols to generate sky fringe-rate centers and widths for.
+        Default is None -> use all keys in self.data.
+    frate_standoff: float, optional
+        Additional fringe-rate standoff in mHz to add to Omega_E b_{EW} nu/c for fringe-rate inpainting.
+        default = 0.0.
+    frate_width_multiplier: float, optional
+        fraction of horizon to fringe-rate filter.
+        default is 1.0
+    min_frate_half_width: float, optional
+        minimum fringe-rate to filter, regardless of baseline length in mHz.
+        Default is 0.025
+
+    Returns
+    -------
+    frate_centers: DataContainer object,
+        DataContainer with the center fringe-rate of each baseline in keys in units of mHz.
+    frate_half_widths: DataContainer object
+        DataContainer with the half widths of each fringe-rate window around the frate_centers in units of mHz.
+
+    """
+    if keys is None:
+        keys = uvd.get_antpairpols()
+    antpos, antnums = uvd.get_ENU_antpos()
+    sinlat = np.sin(np.abs(uvd.telescope_location_lat_lon_alt[0]))
+    frate_centers = {}
+    frate_half_widths = {}
+
+    # compute maximum fringe rate dict based on baseline lengths.
+    for k in keys:
+        ind1 = np.where(antnums == k[0])[0][0]
+        ind2 = np.where(antnums == k[1])[0][0]
+        blvec = antpos[ind1] - antpos[ind2]
+        blcos = blvec[0] / np.linalg.norm(blvec[:2])
+        if np.isfinite(blcos):
+            frateamp_df = np.linalg.norm(blvec[:2]) / SDAY_KSEC / SPEED_OF_LIGHT * 2 * np.pi
+            # set autocorrs to have blcose of 0.0
+
+            if blcos >= 0:
+                max_frate_df = frateamp_df * np.sqrt(sinlat ** 2. + blcos ** 2. * (1 - sinlat ** 2.))
+                min_frate_df = -frateamp_df * sinlat
+            else:
+                min_frate_df = -frateamp_df * np.sqrt(sinlat ** 2. + blcos ** 2. * (1 - sinlat ** 2.))
+                max_frate_df = frateamp_df * sinlat
+
+            min_frate = np.min([f0 * min_frate_df for f0 in uvd.freq_array[0]])
+            max_frate = np.max([f0 * max_frate_df for f0 in uvd.freq_array[0]])
+        else:
+            max_frate = 0.0
+            min_frate = 0.0
+
+        frate_centers[k] = (max_frate + min_frate) / 2.
+        frate_centers[utils.reverse_bl(k)] = -frate_centers[k]
+
+        frate_half_widths[k] = np.abs(max_frate - min_frate) / 2. * frate_width_multiplier + frate_standoff
+        frate_half_widths[k] = np.max([frate_half_widths[k], min_frate_half_width])  # Don't allow frates smaller then min_frate
```

Comment:
I think there are several good reasons to be ouputing and keeping track of the half-width and center of the filter instead of min/max.

First, it is to use a similar API as delay filtering where `min_dly` means the minimum half_width of the delay filter (the current name is misleading, it does not mean the minimum delay of the filter because that is the negative of that value provided while we are centered at zero). lmk if you think we should refactor the delay-filter API and use something like min_dly_half_width instead to make this name less confusing.

Second, using center and width lets us easily perform non-zero centered fringe-rate filtering by using the Fourier shift trick where we just divide the data by the center fringe-rate and then perform a zero-centered filter using the half-width. 

Third, the half-width is useful to output and carry around for auxiliary calculations (like error bars) because it is the factor that is really important in determining the effective averaging time and temporal coherence of the filtered data.

Reply:
I see. FYI--`min_dly` is not meant to signify the minimum filtering delay, as its docstring explains it sets a lower bound on the |tau| filter window (although its explanation can certainly be improved). Your wording in the docstring here actually said "the minimum frate to filter", hence the confusion, but I think it is more clear now. Although I actually think equating it to `min_dly in the delay filter` is actually more confusing b/c even though they are both a half-width, they actually have different implications b/c there is no `center_dly`, so min_dly is an absolute parameter if you will whereas min_frate_half_width is relative. I think the doc is good to go if you just get rid of that line. thanks for humoring me on this

Reply:
Sounds good. I've removed the mention of the min_dly equivalence in the docstring.